pkg/services/thumbs/crawler.go

```diff
@@ -139,16 +144,27 @@ func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme mode
 	}
 	r.broadcastStatus()
 
-	tlog.Info("Starting dashboard crawler", "dashboardsToCrawl", len(items))
-
-	// create a pool of workers
-	for i := 0; i < r.threadCount; i++ {
-		go r.walk()
+	r.logger.Info("Starting dashboard crawler", "dashboardsToCrawl", len(items))
 
-		// wait 1/2 second before starting a new thread
-		time.Sleep(500 * time.Millisecond)
-	}
+	go func() {
+		var wg sync.WaitGroup
+		wg.Add(r.threadCount)
+		// create a pool of workers
+		for i := 0; i < r.threadCount; i++ {
+			go func() {
+				defer wg.Done()
+				r.walk()
+			}()
+			// wait 1/2 second before starting a new thread
+			// TODO: why we are sleeping here?
+			time.Sleep(500 * time.Millisecond)
+		}
+		wg.Wait()
+		r.walkFinished()
+		r.broadcastStatus()
+	}()
 
+	// TODO: what we broadcasting here?
```

Comment:
initial status, ie. running = started

Reply:
But there is one more broadcast above, before starting goroutines

Reply:
oh thats right, didn't notice it - removed